main.js

```diff
@@ -0,0 +1,141 @@
+/* eslint no-param-reassign: ["error", { "props": false }] */
+const boxes = document.querySelectorAll('.box');
+
+const Player = (name, symbol) => ({ name, symbol });
```

Comment:
To simplify our `board` logic we can make sure the `Player` factory has the responsibility of selecting indices. As it is now all we're doing with Player is simply assigning name and symbol, which for that purpose it might be unnecessary to create a Player factory.

Reply:
We are not exactly sure about which method do you refer here. But if it's the `addMark` method then we thinks it's better to be inside `Board` rather than `Player`, because that way the `Player` factory function should call and make a change inside the `Board` factory function. Which is not a good practice in OOP, right?

Reply:
Well acknowledged @jeton-th! :) 

Now the question in my mind here is that, the player object appears to not have any other purpose but just to have a name and symbol. It isn't actually doing anything significant for our application. In that case we might not need to have a factory for player and just create variables in the game flow that represent players 1 and 2. 

What I had in mind is that our `player` object should have a method that uses the `index` / `square` chosen by the user (thru clicking) as an input and then using that index to call on the appropriate `board` method.

Yes, the board should still have a method that updates a particular square a player has chosen. But let's ask the question - where should that index come from? From how it's happening right now, that index comes from the user and directly is used in the board. This then necessitates your `addMark` to also have to take in player as an argument. And this also meant you need to have such a conditional in that method. I was thinking simply more like this:

```javascript
// player factory

const turn = (index) => {
   board.addMark(index, sign);
   this.turns.push(index);
}

// board factory

const addMark = (index, symbol) => board[index] = symbol;
```

Note that in the player factory above I included a line `this.turns(index)`. This may also help you simplify your checkWinner method (see my comment there below)